### ISSUE TITLE
ApacheDS LDAP Server Setup not working

### ISSUE THREAD

**Issue by x44** [dcm4che/dcm4chee-arc-light#452]
Windows 7 64
apacheds-2.0.0-M23
ApacheDirectoryStudio-2.0.0.v20161101-M12
dcm4chee-arc-5.7.2-h2

I also tried this with apacheds-2.0.0-M20 but with equal results.

1) do steps 1. - 4. as described [here](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Installation#apache-ds-200).

2) exit ApacheDirectoryStudio

3) stop ApacheDS. Note that "apacheds.bat default stop" does not terminate the running ApacheDS. Only Ctrl+C terminates the running ApacheDS. If i omit step 4. as described [here](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Installation#apache-ds-200) then "apacheds.bat default stop" does terminate the running ApacheDS.

4) start ApacheDS

5) start ApacheDirectoryStudio

6) create new LDAP Connection in ApacheDirectoryStudio as described in step 1. [here](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Installation#import-default-configuration-into-ldap-server). Note that the port has to be changed to 10389. This connection fails with:
```
Fehler beim Öffnen der Verbindung
 - [LDAP: error code 49 - INVALID_CREDENTIALS: Bind failed: null]
java.lang.Exception: [LDAP: error code 49 - INVALID_CREDENTIALS: Bind failed: null]
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.checkResponse(DirectoryApiConnectionWrapper.java:1268)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.access$9(DirectoryApiConnectionWrapper.java:1236)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper$2.run(DirectoryApiConnectionWrapper.java:440)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.runAndMonitor(DirectoryApiConnectionWrapper.java:1163)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.doBind(DirectoryApiConnectionWrapper.java:449)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.bind(DirectoryApiConnectionWrapper.java:295)
	at org.apache.directory.studio.connection.core.jobs.OpenConnectionsRunnable.run(OpenConnectionsRunnable.java:114)
	at org.apache.directory.studio.connection.core.jobs.StudioConnectionJob.run(StudioConnectionJob.java:109)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

[LDAP: error code 49 - INVALID_CREDENTIALS: Bind failed: null]
```

7) since this connection fails, i try to continue with the LDAP connection created in step 2. [here](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Installation#apache-ds-200)

8) import $DCM4CHEE_ARC/ldap/init-baseDN.ldif -> OK
9) import $DCM4CHEE_ARC/ldap/init-config.ldif -> OK
10) import $DCM4CHEE_ARC/ldap/default-config.ldif -> FAILS:
```
Fehler beim LDIF Importieren
 - [LDAP: error code 32 - NO_SUCH_OBJECT: failed for MessageType : ADD_REQUES
java.lang.Exception: [LDAP: error code 32 - NO_SUCH_OBJECT: failed for MessageType : ADD_REQUEST
Message ID : 25
    Add Request :
Entry
    dn[n]: cn=syslog,dicomDeviceName=keycloak,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org
    objectClass: dicomNetworkConnection
    objectClass: dcmNetworkConnection
    dcmClientBindAddress: 0.0.0.0
    dicomHostname: localhost
    cn: syslog
    dcmProtocol: SYSLOG_UDP
: ERR_251_PARENT_NOT_FOUND Parent dicomDeviceName=keycloak,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org not found]
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.checkResponse(DirectoryApiConnectionWrapper.java:1268)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.access$9(DirectoryApiConnectionWrapper.java:1236)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper$6.run(DirectoryApiConnectionWrapper.java:918)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.runAndMonitor(DirectoryApiConnectionWrapper.java:1163)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.checkConnectionAndRunAndMonitor(DirectoryApiConnectionWrapper.java:1099)
	at org.apache.directory.studio.connection.core.io.api.DirectoryApiConnectionWrapper.createEntry(DirectoryApiConnectionWrapper.java:940)
	at org.apache.directory.studio.ldapbrowser.core.jobs.ImportLdifRunnable.importLdifRecord(ImportLdifRunnable.java:465)
	at org.apache.directory.studio.ldapbrowser.core.jobs.ImportLdifRunnable.importLdif(ImportLdifRunnable.java:272)
	at org.apache.directory.studio.ldapbrowser.core.jobs.ImportLdifRunnable.run(ImportLdifRunnable.java:218)
	at org.apache.directory.studio.connection.core.jobs.StudioConnectionJob.run(StudioConnectionJob.java:109)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

[LDAP: error code 32 - NO_SUCH_OBJECT: failed for MessageType : ADD_REQUEST
Message ID : 25
    Add Request :
Entry
    dn[n]: cn=syslog,dicomDeviceName=keycloak,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org
    objectClass: dicomNetworkConnection
    objectClass: dcmNetworkConnection
    dcmClientBindAddress: 0.0.0.0
    dicomHostname: localhost
    cn: syslog
    dcmProtocol: SYSLOG_UDP
: ERR_251_PARENT_NOT_FOUND Parent dicomDeviceName=keycloak,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org not found]
```

**Comment by x44** [comment]
sorry for beeing impatient, but we are short in time :)

- can you reproduce this problem?
- is this a problem of ApacheDS or DCM4CHEE?
- any workaround?

thanks!

**Comment by gunterze** [comment]
It looks that you Apache DS configuration has a different Bind DN/password than uid=admin,ou=system / secret. Have currently no time to try to reproduce you problem. Use OpenLDAP or just the dockerized version, if you don't want to wait.

**Comment by x44** [comment]
I am using uid=admin,ou=system / secret, otherwise the connection used for the first steps would fail.

**Comment by gunterze** [comment]
`cn=admin,dc=dcm4che,dc=org` is the default Bind DN using OpenLDAP!

**Comment by mikivan1980** [comment]
There were similar problems with the file - dcm4chee-archive.ldif. Ldap parser did not accept some of the strings as valid. This tool http://jxworkbench.com/  provides a detailed log.

**Comment by gunterze** [comment]
Also after fix of #414 ?

**Comment by mikivan1980** [comment]
Wow, wow... Yes, true, no error occurs if   \"expirationdate\"\\s*:\\s*\"([0-9]{8})\"   
I did not understand the format, simply deleted this expr..
